modules/k6/k6_test.go

```diff
@@ -0,0 +1,58 @@
+package k6
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+)
+
+func TestK6(t *testing.T) {
+	testCases := []struct {
+		title  string
+		script string
+		expect int
+	}{
+		{
+			title:  "Passing test",
+			script: "pass.js",
+			expect: 0,
+		},
+		{
+			title:  "Failing test",
+			script: "fail.js",
+			expect: 108,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.title, func(t *testing.T) {
+			ctx := context.Background()
+
+			absPath, err := filepath.Abs(filepath.Join("scripts", tc.script))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			container, err := RunContainer(ctx, WithTestScript(absPath))
```

Comment:
Could you add the cache here? It will demonstrate the benefits of using it, also speeding up the build (I did try locally)
```suggestion
			container, err := RunContainer(ctx, WithTestScript(absPath), WithCache("k6-cache"))
```